### PR TITLE
BOP flowers

### PR DESCRIPTION
**Pull request by EnderProyects**
Added chain recipes for BOP flowers

https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13753#issue-1763745419

### CODE REVIEW

**Comment by combusterf** [comment]
This almost entirely removes the exploration aspect and makes all other approaches useless.

No thanks.

**Comment by wlhlm** [comment]
Please don't just close pull requests without discussion when other devs [voiced](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13753) being fine with the changes.

**Comment by EnderProyects** [comment]
I oredicted a vine wrong, correcting it ( idk why it worked at my pc)

File: src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java [hunk 1]
```diff
@@ -187,6 +191,30 @@ public static void applyPatches() {
...
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerBurningBlossom").get(0), "flowerLavender", 16000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerLavender").get(0), "flowerGoldenrod", 8000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerGoldenrod").get(0), "flowerBluebell", 8000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerBluebell").get(0), "flowerMinersDelight", 8000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerMinersDelight").get(0),"flowerIcyIris", 8000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerIcyIris").get(0), "flowerRose", 8000));
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerRose").get(0), "flowerClover", 8000));
```
Comment (combusterf) [inline src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: There are much smarter ways to do this.

**Review by combusterf — CHANGES_REQUESTED** [review]
General reject:
- Removes the existing exploration balance in its entirety.
- Includes undiscussed features.
- Generally uncreative.

**Comment by Alastors** [comment]
> General reject:
> 
> * Removes the existing exploration balance in its entirety.
> * Includes undiscussed features.
> * Generally uncreative.

could you clarify more what you mean by the undiscussed features?

**Comment by Alastors** [comment]
Also I'm just going to put this out there, bees already generate these flowers

**Comment by EnderProyects** [comment]
> Also I'm just going to put this out there, bees already generate these flowers

I dont see something bad in having another way to generate them and that you can controll the result

**Comment by Alastors** [comment]
> > Also I'm just going to put this out there, bees already generate these flowers
> 
> I dont see something bad in having another way to generate them and that you can controll the result

it wasn't bad, it was merely an observation

**Comment by EnderProyects** [comment]
I dont know why is failing here, it works on my side.
The only thing that I can think about if is if the new recipes should be done here or in the botania code. @combusterf ?

**Comment by Alastors** [comment]
> I dont know why is failing here, it works on my side. The only thing that I can think about if is if the new recipes should be done here or in the botania code. @combusterf ?

clear your ide's cache lad

File: src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java [hunk 2]
```diff
@@ -142,6 +148,8 @@ public static void applyPatches() {
         ModManaAlchemyRecipes.slimeballCactusRecipes = new ArrayList<>();
         ModManaAlchemyRecipes.slimeballCactusRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Items.slime_ball), new ItemStack(Blocks.cactus), 1200));
         ModManaAlchemyRecipes.slimeballCactusRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.cactus), new ItemStack(Items.slime_ball), 1200));
+        //ModManaAlchemyRecipes.slimeballCactusRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("cropCactusfruit").get(0),"cropSmallCactus", 8000));
+        //ModManaAlchemyRecipes.slimeballCactusRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("cropSmallCactus").get(0),"cropCactusfruit", 8000));
```
Comment (combusterf) [inline src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: Don't leave your test code in place

File: src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java [hunk 3]
```diff
@@ -111,6 +114,9 @@ public static void applyPatches() {
         ModManaAlchemyRecipes.swampFolliageRecipes = new ArrayList<>();
         ModManaAlchemyRecipes.swampFolliageRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.waterlily), new ItemStack(Blocks.vine), 320));
         ModManaAlchemyRecipes.swampFolliageRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.vine), new ItemStack(Blocks.waterlily), 320));
+        //ModManaAlchemyRecipes.swampFolliageRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("cropIvy").get(0),"cropFloweringVines", 8000));
+        //ModManaAlchemyRecipes.swampFolliageRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("cropFloweringVines").get(0),"cropThornVines", 8000));
+        //ModManaAlchemyRecipes.swampFolliageRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("cropThornVines").get(0),"cropIvy", 8000));
```
Comment (combusterf) [inline src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: Don't leave your test code in place

File: src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java [hunk 4]
```diff
@@ -186,7 +194,32 @@ public static void applyPatches() {
         ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.double_plant, 1, 4), new ItemStack(Blocks.double_plant, 1, 1), 400));
         ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.double_plant, 1, 5), new ItemStack(Blocks.double_plant, 1, 4), 400));
         ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(new ItemStack(Blocks.yellow_flower), new ItemStack(Blocks.double_plant, 1, 5), 400));
-
+            if(Loader.isModLoaded("BiomesOPlenty")){
+        ModManaAlchemyRecipes.flowersRecipes.add(BotaniaAPI.registerManaAlchemyRecipe(OreDictionary.getOres("flowerClover").get(0),"flowerSwamp", 8000));
```
Comment (combusterf) [inline src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: Indentation error

**Review by combusterf — CHANGES_REQUESTED** [review]
Quality has only gotten worse

File: src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java (hunk 4 above)
Comment (combusterf) [reply src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: Please do not resolve unfixed issues.

Comment (EnderProyects) [reply src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: I thought I changed this file yesterday wtf

Comment (EnderProyects) [reply src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: Yes, is fixed in the last one

Comment (combusterf) [reply src/main/java/net/fuzzycraft/botanichorizons/patches/ManaInfusionPatches.java]: You unindented the first line, but you still haven't indented everything within {...}

**Comment by Alastors** [comment]
You should consider using IntelliJ if you aren't already

**Pull request closed without merging**